protos/sources/vega/snapshot/v1/snapshot.proto

```diff
@@ -836,6 +840,48 @@ message LiquidityScore {
   string party_id = 2;
 }
 
+message LiquidityV2Provisions {
+  // Market ID these liquidity parameters apply to. Required for key generation
+  string market_id = 1;
+
+  repeated vega.LiquidityProvision liquidity_provisions = 2;
```

Comment:
I don't think this makes sense anymore. We should use a LiquidityProvisionV2 which doesn't have all of the reference, shape nonsense.

Reply:
we should keep the LiquidityProvision type as it is now, in the state serialization not settign the shap is no cost. And we keep the benefit of not having to change the APIs anywhere by keep the same types.

Reply:
this type is already v2, what's the point in building new types and reusing existing types that will never be valid for it?

Reply:
I'm not sure I get it, there's no new LiquidityProvision type.

Reply:
```
message **LiquidityV2Provisions** {
  // Market ID these liquidity parameters apply to. Required for key generation
  string market_id = 1;

  repeated vega.LiquidityProvision liquidity_provisions = 2;
```

LiquidityV2Provisions is a new type, the existing type vega.LiquidityProvision will *never* be right for it, so why use it?

Reply:
why confuse the API with fields that can never be non empty?

Reply:
well, the only thing which is not right is the fields for the shapes. But they still exists in the historic tables etc. we can make them nullable now and deprecate it in the long term.

Reply:
this is the *snapshot* what it's got to do with any of that? anyways, doesn't really matter, just seems unclean.

Reply:
Doesn't have much to do with it yes, but it's just adding one more type, with convertion at some point from one to the other, which is unecessary right now I'd say.